### DISCUSSION TITLE
Possible to get order of keys in parsed toml?

### DISCUSSION THREAD

**Discussion by rszyma** [pelletier/go-toml#939] [Questions] [answered]
Golang `map` doesn't preserve order of insertion, so when I parse a toml file to a struct containing map the keys are shuffled.

Is it possible in go-toml to somehow get an array of keys that will complement the lack of ordering in `map`?

**Comment by pelletier** [comment, marked as answer]
This answer may be helpful to you: https://github.com/pelletier/go-toml/discussions/801#discussioncomment-7083586

In brief, you can use the unstable API to parse the document directly. It's not a great answer, but I think it's the best available option until https://github.com/pelletier/go-toml/issues/873 is fully supported.

**Reply by rszyma** [reply]
The example you've provided in #801 is very helpful, thank you!
